src/main/java/tdl/client/oldrunner/CombinedClient.java

```diff
@@ -0,0 +1,77 @@
+package tdl.client.oldrunner;
+
+import tdl.client.Client;
+import tdl.client.ProcessingRules;
+import tdl.client.abstractions.UserImplementation;
+import tdl.client.actions.ClientAction;
+
+import java.util.Map;
+import java.util.function.Consumer;
+
+import static tdl.client.actions.ClientActions.publish;
+
+
+public class CombinedClient {
+    private HttpClient httpClient;
+    private String hostname;
+    private String username;
+    private Consumer<String> printer;
+
+    public CombinedClient(String journeyId, boolean useColours, String hostname, String username, Consumer<String> printer) {
+        this.hostname = hostname;
+        this.username = username;
+        this.printer = printer;
+        httpClient = new HttpClient(hostname, journeyId, useColours);
+    }
+
+    public boolean checkStatusOfChallenge() throws HttpClient.ServerErrorException, HttpClient.OtherCommunicationException, HttpClient.ClientErrorException {
+        String journeyProgress = httpClient.getJourneyProgress();
+        printer.accept(journeyProgress);
```

Comment:
`printer.accept` does not read nicely
I would not use the Consumer interface (very Java specific) and use a proper Interface instead.